launch_testing/launch_testing/loader.py

```diff
@@ -231,6 +243,43 @@ def _partially_bind_matching_args(unbound_function, arg_candidates):
 
 
 def _give_attribute_to_tests(data, attr_name, test_suite):
+
+    # Accessing the proc_info, proc_output, etc. . . objects via self.proc_info
+    # will be removed in a future release.  Wrap these objects in a way that produces a warning
+    # when they are used
+    class _warn_wrapper:
+
+        def __init__(self, wrapped):
+            self.__warned = False
+            self.__wrapped = wrapped
+
+        def __getattr__(self, attr):
+            self._warn_once()
+            return getattr(self.__wrapped, attr)
+
+        def __getitem__(self, key):
+            self._warn_once()
+            return self.__wrapped[key]
+
+        def __iter__(self):
+            self._warn_once()
+            return self.__wrapped.__iter__()
+
+        def _warn_once(self):
+
+            if self.__warned:
+                return
+
+            _logger.warning(
+                'Automatically adding attributes like self.proc_info and self.proc_output '
+                'to the test class will be deprecated in a future release.  '
+                'Instead, add proc_info or proc_output to the test method argument list to '
+                'access the test object you need'
+            )
+            self.__warned = True
+
+    data = _warn_wrapper(data)
```

Comment:
@pbaughman meta: wouldn't it be simpler to turn inject attributes into properties that warn on access? Instead of trying to cover every possible way these attribute may be used I mean.

Reply:
@hidmic I will try - I'm not sure what it takes to dynamically add a property to a class but I'm sure I can figure it out

Reply:
See [here](https://docs.python.org/3.6/howto/descriptor.html#properties).